jumpscale/clients/btc_alpha/btc_alpha.py

```diff
@@ -0,0 +1,195 @@
+        :param kwargs: Filters (Optional): pair, limit
+        :return: exchanges array
+        Example: [{'id': 6030, 'price': '839.36000000', 'pair': 'BTC_USD', 'type': 1, 'timestamp': 1483705817.735508,
+                'amount': '0.00281167'}, ....]
+        """
+        return self._query("get", "v1/exchanges/", params=kwargs)
+
+    def get_own_exchanges(self, **kwargs):
+        """Returns only own exchanges
+        :param kwargs: see get_exchanges()
+        :return: see get_exchanges()
+        """
+        return self._query("get", "v1/exchanges/own/", params=kwargs, auth=True)
+
+    def get_own_sell_exchanges(self, **kwargs):
+        kwargs["type"] = "sell"
+        return self._query("get", "v1/exchanges/own/", params=kwargs, auth=True)
+
+    def get_own_buy_exchanges(self, **kwargs):
+        kwargs["type"] = "buy"
+        return self._query("get", "v1/exchanges/own/", params=kwargs, auth=True)
+
+    def get_deposits(self):
+        """Returns all made deposits
+        :return: deposits array
+        Example [{'timestamp': 1485363039.18359, 'id': 317, 'currency': 'BTC', 'amount': '530.00000000'}, ... ]
+        """
+        return self._query("get", "v1/deposits/", auth=True)
+
+    def get_withdraws(self, **kwargs):
+        """Returns all made withdraws
+        :param kwargs: currency_id, status
+        :return: withdraws array
+        Example: [{'id': 403, 'timestamp': 1485363466.868539, 'currency': 'BTC', 'amount': '0.53000000', 'status': 20} ...]
+        """
+
+        return self._query("get", "v1/withdraws/", params=kwargs, auth=True)
+
+    def get_orderbook(self, pair, **kwargs):
+        """Return sell and buy orders for specified pair
+        :param kwargs: limit_bids, limit_asks, group
+        :return: object {'bids': Array, 'asks': Array}
+        Example: {
+            'bids': [{'price': 911.519, 'id': 44667, 'amount': 0.000446, 'timestamp': 1485777324.410015}],
+            'asks': [{'price': 911.122, 'id': 44647, 'amount': 0.001233, 'timestamp': 1485777124.415542}]
+        }
+        """
+
+        return self._query("get", "v1/orderbook/{}/".format(pair), params=kwargs)
+
+    def get_charts(self, pair, type, **kwargs):
+        """Return data for chart candles
+        :param pair: Pair name
+        :param type: Type of candles ('1','5','15','30','60','240','D')
+        :param kwargs: since, until, limit
+        :return: array of bars
+        Example [
+            {'volume': 0.262929, 'high': 912.236, 'low': 910.086, 'close': 911.915, 'time': 1485777600, 'open': 910.424},
+            {'volume': 2.16483814, 'high': 911.519, 'low': 909.432, 'close': 910.424, 'time': 1485774000, 'open': 909.433}
+        ]
+        """
+        return self._query("get", "charts/{}/{}/chart".format(pair, type), params=kwargs)
+
+    def _query(self, method, base_url, params=None, data=None, auth=False):
+        request_conf = {"method": method, "url": self._url + base_url, "allow_redirects": True}
+
```

Comment:
I believe this method can be rewritten to:
```suggestion
def _query(self, method, path, params=None, data=None, auth=False):
     url = f"{self._url}/{path}"
     headers = data and self._get_headers(data)
     response = requests.request(method=method, url=f"{self._url}/{path}", params=params, data=data, auth=auth, headers=headers)
```

Reply:
done and some other cleaning, thanks